chat-server/src/routes/conversations/addMessageToConversation.ts

```diff
@@ -357,64 +296,70 @@ export function makeAddMessageToConversationRoute({
             reqId,
             message: "Only sending vector search results to user",
           });
-          answer = {
+          const answer = {
             role: "assistant",
             content: conversationConstants.LLM_NOT_WORKING,
           } satisfies OpenAiChatMessage;
-          answerContent = answer.content;
+          return answer.content;
         }
-      }
+      })();
 
       if (!answerContent) {
-        throw new Error("No answer content");
+        throw makeRequestError({
+          httpStatus: 500,
+          message: "No answer content",
+        });
       }
 
+      // --- SAVE QUESTION & RESPONSE ---
       const { assistantMessage } = await addMessagesToDatabase({
         conversations,
-        conversationId,
+        conversation,
         originalUserMessageContent: message,
         preprocessedUserMessageContent,
         assistantMessageContent: answerContent,
         assistantMessageReferences: references,
-        userMessageEmbedding: embedding,
+        userMessageEmbedding: queryEmbedding,
       });
 
       const apiRes = convertMessageFromDbToApi(assistantMessage);
-      if (shouldStream) {
-        dataStreamer.streamData({
-          type: "finished",
-          data: apiRes.id,
-        });
-        dataStreamer.disconnect();
-        return;
-      } else {
-        res.status(200).json(apiRes);
+
+      if (!shouldStream) {
+        return res.status(200).json(apiRes);
       }
-    } catch (err) {
-      logRequest({
-        reqId,
-        message: "An unexpected error occurred: " + JSON.stringify(err),
-        type: "error",
+
+      dataStreamer.streamData({
```

Comment:
i think this stuff could be inside this `if` statement https://github.com/mongodb/chatbot/pull/199/files#diff-ad16696d33139a2879677f2ca6d0ef3291daf4386dc13ecfad6b735809392134R261

Reply:
We need the apiRes.id which is only generated after storing the message in the database. This could probably be refactored further but I'd rather iterate on this PR later.